src/contrib/cache_req.rst

```diff
@@ -0,0 +1,290 @@
+.. contents:: Table of Contents
+    :local:
+
+Cache Req
+=========
+
+The ``cache req`` (request) interface provides an internal API to request identity
+information from SSSD, either directly from the local SSSD cache or retrieving refreshed data
+from the data provider. The SSSD NSS responder is the primary consumer of the interface,
+however ``cache_req`` also refreshes information which may be needed by the PAM
+responder, or provide user information attributes over the SSSD infopipe.
+
+.. mermaid::
+
+	graph LR
+		A[Application] --> B(glibc-NSS)
+		B --> C(sssd_nss)
+		C --> D{cache_req}
+		D --> E(cache)
+		E <--> F(backend)
+		F --> E
+
+For informational purposes, we will follow the path of a typical ``getpwnam()`` glibc
+call (`id username`) to understand how it interacts with the cache req interface. You
+can follow along the logic by analyzing the ``/var/log/sssd/sssd_nss.log`` file with
+SSSD debugging enabled.
+
+NSS responder request setup
+---------------------------
+
+In preparation for setting up a request, The NSS responder calls into the function
+``nss_cmd_getpwnam()``. This function is a wrapper of the generic function
+``nss_getby_name()`` with the  argument ``cache_req_type`` enum ``CACHE_REQ_USER_BY_NAME``.
+The cache request types can be understood as a mapping from the equivalent NSS
+library interface lookup functions. ``getgetnam()`` sets ``CACHE_REQ_GROUP_BY_NAME``,
+``nss_cmd_getgrgid(``) sets ``CACHE_REQ_GROUP_BY_ID()``, etc...
```

Comment:
I think I would put here full definition of `enum cache_req_type` in some scrollable box or table with some mappings in between functions and `cache_req` types.